graph/graphtest/graphtest.go

```diff
@@ -528,63 +534,38 @@ func TestLoadTypedQuads(t testing.TB, gen DatabaseFunc, conf *Config) {
 	require.Equal(t, int64(7), qs.Size(), "Unexpected quadstore size")
 }
 
-// Came across a bug where a time.Time quad returned by iterator.NewAnd could not be deleted even though it exists.
-// dennwc said that this has something to do with protobuf encoding dropping timezone info which is used in hash/node id generation.
-func TestReadTimeQuadsViaIterator(t testing.TB, gen DatabaseFunc, conf *Config) {
-	qs, opts, closer := gen(t)
-	defer closer()
-
-	w := MakeWriter(t, qs, opts)
-	now := time.Now()
-	subject := quad.String("timequad")
-	values := []quad.Quad{
-		{subject, quad.String("has_unixtime"), quad.Int(now.UnixNano()), nil},
-		{subject, quad.String("has_time"), quad.Time(now), nil},
-	}
-	err := w.AddQuadSet(values)
-	require.Nil(t, err)
-
-	for _, v := range values {
-		it := iterator.NewAnd(
-			qs,
-			qs.QuadIterator(quad.Subject, qs.ValueOf(subject)),
-			qs.QuadIterator(quad.Predicate, qs.ValueOf(v.Predicate)),
-		)
-
-		for it.Next() {
-			assert.Equal(t, v, qs.Quad(it.Result()), "Quads do not match")
-		}
-
-		it.Close()
-	}
-}
-
-// Came across a bug where a time.Time quad returned by iterator.NewAnd could not be deleted even though it exists.
+// Came across a bug where a time.Time quad could not be deleted even though it exists.
 // dennwc said that this has something to do with protobuf encoding dropping timezone info which is used in hash/node id generation.
-func TestRemoveTimeQuadsViaIterator(t testing.TB, gen DatabaseFunc, conf *Config) {
+func TestRemoveQuadsViaIterator(t testing.TB, gen DatabaseFunc, conf *Config) {
+	// Locals.
 	qs, opts, closer := gen(t)
 	defer closer()
 
+	// Create writer and test-value set.
 	w := MakeWriter(t, qs, opts)
 	now := time.Now()
-	subject := quad.String("timequad")
+	subject := quad.IRI("timequad")
 	values := []quad.Quad{
 		{subject, quad.String("has_unixtime"), quad.Int(now.UnixNano()), nil},
 		{subject, quad.String("has_time"), quad.Time(now), nil},
 	}
 	err := w.AddQuadSet(values)
 	require.Nil(t, err)
 
-	it := iterator.NewAnd(
-		qs,
-		qs.QuadIterator(quad.Subject, qs.ValueOf(subject)),
-	)
+	// Create iterator.
+	it := qs.QuadsAllIterator()
 	defer it.Close()
+	
+	// Check iterator size.
+	count, exact := it.Size()
```

Comment:
True. But line below requires result to be exact. This is not true for some backends.